src/Xenial.Tasty.Core/Commander.Tasty.cs

```diff
@@ -117,43 +137,69 @@ private async Task ReportSummary(IEnumerable<TestCaseResult> testCases)
             }
         }
 
+        internal async Task<TastyServer?> ConnectToRemote(System.Uri connectionString, CancellationToken cancellationToken = default)
+        {
+            if (TransportStreamFactories.TryGetValue(connectionString.Scheme, out var factory))
+            {
+                var result = await factory(connectionString, cancellationToken).ConfigureAwait(false);
+                var streamTask = result();
+
+                disposables.Add(streamTask);
+                var stream = await streamTask.ConfigureAwait(false);
+
+                tastyServer = new TastyServer();
+
+                tastyServer.RegisterReporter(Report);
+                tastyServer.RegisterReporter(ReportSummary);
+                tastyServer.EndTestPipelineSignaled += () => running = false;
+
+                jsonRpc = JsonRpc.Attach(stream, tastyServer);
+                disposables.Add(jsonRpc);
+                running = true;
+                return tastyServer;
+            }
+            return null;
+        }
+
+        public async IAsyncEnumerable<TestCaseResult> WaitForResults()
```

Comment:
Currently there is no way for timeouts. We should address this in another issue.